x-pack/plugins/security_solution/server/endpoint/routes/resolver/queries/base.ts

```diff
@@ -17,7 +17,7 @@ import { MSearchQuery } from './multi_searcher';
  * @param T the structured return type of a resolver query. This represents the type that is returned when translating
  * Elasticsearch's SearchResponse<ResolverEvent> response.
  */
-export abstract class ResolverQuery<T> implements MSearchQuery {
+export abstract class ResolverQuery<T, R = ResolverEvent> implements MSearchQuery {
```

Comment:
The `R` here sets the stage for being able to limit the `_source` of a queries response. This will be in a follow up PR. Limit the source will be a performance improvement for querying for the start events for the children.